compiler/src/dotty/tools/dotc/typer/Implicits.scala

```diff
@@ -610,6 +610,8 @@ trait ImplicitRunInfo:
         else
           partSeen += t
           t.dealias match
+            case dealias if t ne dealias =>
```

Comment:
I don't see any motivation for this code, and without it the change seems unsystematic. There is no wildApprox anywhere else in this function. Why introduce it here?

Reply:
I have no idea how to find a better way to put this check. The only thing I've discovered is that this type is not going through `wildApprox` when I guess it should.

I'll close this MR then.